packages/eslint-plugin/src/rules/unified-signatures.ts

```diff
@@ -213,6 +213,9 @@ export default util.createRule<Options, MessageIds>({
         b.typeParameters !== undefined ? b.typeParameters.params : undefined;
 
       if (ignoreDifferentlyNamedParameters) {
+        if (b.params.length === 0) {
+          return false;
+        }
```

Comment:
You need to check that _every_ `b.params[i]` is defined. [Test case](https://deploy-preview-6940--typescript-eslint.netlify.app/play/#ts=5.0.4&sourceType=module&code=GYVwdgxgLglg9mABMAFANwFyIEZzgGwFMBDMAGkWK1wJLAEoswQBbbQgJwG4AoUSWAmTpqeIqUaIAzlA4wwAc15A&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6WJygM0sQBNaySgHMmAQ3yxoKdGADa4bDkTRoAe2iQANIqVYl2SCKYbEAEW5cViFvEIA5MQFt+ABTHRnifCtQZ8cIi62AC+wQC6imEhQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA)

Reply:
@Josh-Cena 
I fixed it in 47286f7